app/controllers/users_controller.rb

```diff
@@ -2,10 +2,15 @@
 
 class UsersController < ApplicationController
   def show
-    if params[:nickname] == current_user.nickname
-      @posts = current_user.posts.order(published_at: :desc)
-    else
-      redirect_to root_path
-    end
+    @user = User.find_by(nickname: params[:nickname])
+    @posts = @user.posts.order(published_at: :desc) if @user
+
+    return if @user
+
+    redirect_to root_path, alert: 'User not found'
```

Comment:
Let's re-order this so it's clearer
```suggestion
    redirect_to root_path, alert: 'User not found' unless @user

    @posts = @user.posts.order(published_at: :desc)
```